### PR TITLE
NMS-12949: Persist BMP messages

### PR DESCRIPTION
**Pull request by cgorantla**
This issue handles all the basic tables like collectors / routers / peers / prefixes / bmp attributes.

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12949

### CODE REVIEW

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+                            // Upon initial router message in INIT/FIRST state,  update all corresponding peer state to down.
+                            boolean state = !router.getAction().equals(Router.Action.TERM.value);
+                            if (connections == 0 && state) {
+                                router.getBmpPeers().forEach(bmpPeer -> {
+                                    if (bmpPeer.getTimestamp().getTime() < router.getTimestamp().getTime()) {
+                                        bmpPeer.setState(false);
+                                    }
+                                });
+                            }
+                            Integer count = state ? ++connections : --connections;
```
Comment (fooker) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpMessagePersister.java:171]: Lovin' it

**Review by fooker — CHANGES_REQUESTED** [review]
Looks good.
On question: How do we configure the persistence RRAs and does this resonate with the stats update strategy?

**Review by cgorantla — COMMENTED** [review]
> Looks good.
> On question: How do we configure the persistence RRAs and does this resonate with the stats update strategy?

Updated RRAs in config. Currently stats get updated like all other telemetry metrics.

File: core/schema/src/main/liquibase/27.0.0/changelog.xml [hunk 2]
```diff
@@ -161,17 +161,14 @@
         <column name="hash_id" type="varchar(36)">
           <constraints nullable="false"/>
         </column>
-        <!-- Represents state Up/Down, default is Down -->
-        <column name="state" type="boolean" defaultValue="false">
+        <column name="state" type="varchar(8)">
```
Comment (fooker) [inline core/schema/src/main/liquibase/27.0.0/changelog.xml:164]: Do we need a default value here, too?

File: features/telemetry/protocols/bmp/persistence/impl/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/persistence/impl/BmpGlobalIpRibDaoImpl.java [hunk 3]
```diff
@@ -36,7 +36,7 @@
 import org.opennms.netmgt.telemetry.protocols.bmp.persistence.api.BmpGlobalIpRib;
 import org.opennms.netmgt.telemetry.protocols.bmp.persistence.api.BmpGlobalIpRibDao;
 
-public class BmpGlobalIpRibDaoImpl extends AbstractDaoHibernate<BmpGlobalIpRib, Long> implements BmpGlobalIpRibDao {
+public class    BmpGlobalIpRibDaoImpl extends AbstractDaoHibernate<BmpGlobalIpRib, Long> implements BmpGlobalIpRibDao {
```
Comment (fooker) [inline features/telemetry/protocols/bmp/persistence/impl/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/persistence/impl/BmpGlobalIpRibDaoImpl.java:39]: Oops

**Review by fooker — CHANGES_REQUESTED** [review]
I like it a lot. Especially the `State` handling. Just a few minor nits left.

**Review by fooker — APPROVED** [review]

**Pull request merged**